Dockerfile

```diff
@@ -5,13 +5,15 @@ WORKDIR /usr/src/app
 
 COPY package*.json ./
 
-RUN npm ci --production
-
-USER node
+RUN npm install
```

Comment:
this will end up with all our dev dependencies included in our docker image, but they aren't used at runtime.
This increases the size of the image.

The solution is to use a multi stage image.
First we copy everything, do an `npm ci` and run `tsc`
Then we copy the dist folder across, along with the package and package-lock, and do an `npm ci --production`
the second install is super quick as its cached but wont include the devDependencies section of the package.json

More in depth instructions https://stackoverflow.com/questions/60916271/how-do-i-run-typescript-tsc-before-copy-in-dockerfile-template